os/pm/pm_timer/pm_timer_set.c

```diff
@@ -0,0 +1,293 @@
+
+        /* Case where there are no timers in the list */
+
+        if (g_pmTimer_activeList.head == NULL) {
+                sq_addlast((FAR sq_entry_t *)timer, &g_pmTimer_activeList);
+                pmdbg("timer added in the head of linked list\n");
+        }
+        
+        /* We should add the timer in the sorted position of expiration time */
+
+         else {
+                pm_wakeup_timer_t *curr;
+                pm_wakeup_timer_t *prev;
+                prev = curr = (pm_wakeup_timer_t *)g_pmTimer_activeList.head;
+
+                while (timer->expire_timetick >= curr->expire_timetick && curr->next) {
+                        prev = curr;
+                        curr = curr->next;
+                }
+
+                /* timer should be added before the curr */
+                if (timer->expire_timetick < curr->expire_timetick) {
+                        
+                        if (curr == (pm_wakeup_timer_t *)g_pmTimer_activeList.head) {
+                                /* if the first timer in the list was already running make it active */
+                                curr->status = ACTIVE;
+                                sq_addfirst((FAR sq_entry_t *)timer, &g_pmTimer_activeList);
+                        } else {
+                                sq_addafter((FAR sq_entry_t *)prev, (FAR sq_entry_t *)timer, &g_pmTimer_activeList);
+                        }  
+
+                /* timer expire time is greater than every other timer. 
+                 * timer should be added to the end of the linked list*/
+                } else {
+                        if (!curr->next) {
+                                sq_addlast((FAR sq_entry_t *)timer, &g_pmTimer_activeList);
+                        } else {
+                                sq_addafter((FAR sq_entry_t *)curr, (FAR sq_entry_t *)timer, &g_pmTimer_activeList);
+                        }
+                }
+                pmdbg("timer with id %d added in linked list by using while loop\n", timer->id);
+        }
+        timer->status = ACTIVE;
+}
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+
+/************************************************************************
+ * Name: pm_timer_set
+ *
+ * Description:
+ *   This function adds a wakeup timer in the g_pmTimer_activeList. So that it will be
+ *   invoked just before sleep when needed. It also removes the wakeup timer if
+ *   its already present in the list.
+ *
+ * Parameters:
+ *   id - id of the timer
+ *   timer_interval - expected board sleep duration
+ *
+ * Return Value:
+ *   0 - success
+ *   -1 - error
+ *
+ ************************************************************************/
+
+int pm_timer_set(int id, unsigned int timer_interval)
+{
+        pm_wakeup_timer_t *timer = get_pm_timer(id);
+        DEBUGASSERT(timer);
```

Comment:
I think DEBUGASSERT is too high option here. Return error & debug message is enough....please discuss with your team about this.